File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface BrandService {


	
	

	
	
	/**
	 * 增加
	*/
	public void add(TbBrand brand);
	
	
	/**
	 * 修改
	 */
	public void update(TbBrand brand);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, Integer pageNum, Integer pageSize);

    List<Map> selectBrandList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../brand/search.do?page=' + pageNum + '&rows=' + pageSize, searchEntity);
    }

    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }

    this.dele = function (selectIds) {
        return $http.get('../brand/delete.do?ids=' + selectIds);
    }

    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }

    this.update = function (entity) {
        return $http.post('../brand/update.do',entity);
    }

    this.selectOptionList=function () {
        return $http.get("../brand/selectBrandList.do")
    }
})